cmd/wego/app/add/cmd.go

```diff
@@ -114,11 +145,38 @@ func runCmd(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("error selecting auth method: %w", err)
 	}
 
-	cliRunner := &runner.CLIRunner{}
-	fluxClient := flux.New(osysClient, cliRunner)
-	kubeClient := kube.New(cliRunner)
 	gitClient := git.New(authMethod)
-	logger := logger.NewCLILogger(os.Stdout)
+
+	// If we are NOT doing a helm chart, we want to use a git client with an embedded deploy key
+	if !isHelmChart {
+		authsvc, err := auth.NewAuthService(fluxClient, rawClient, providerName, logger, token)
```

Comment:
This function has numerous shadow variables for `err`.  Consider renaming.

Reply:
I'm going to resolve this, but leave it to you to decide what you want to do.  Again, they are subtle when they appear.